src/Store.js

```diff
@@ -580,31 +651,48 @@ class Store {
   /**
    * Gets records for type of collection from observable
    *
+   * NOTE: We only return records by unique id, this handles a scenario
+   * where the store keeps around a reference to a newly persisted record by its temp uuid.
+   * We can't simply remove the temp uuid reference because other
+   * related models may be still using the temp uuid in their relationships
+   * data object. However, when we are listing out records we want them
+   * to be unique by the persisted id (which is updated after a Model.save)
+   *
    * @method getRecords
    * @param {String} type
    * @return {Array} array of objects
    */
   getRecords (type) {
-    const records = Array.from(this.getType(type).records.values())
-                         .filter(value => value && value !== 'undefined')
-
-    // NOTE: Handles a scenario where the store keeps around a reference
-    // to a newly persisted record by its temp uuid. This is required
-    // because we can't simply remove the temp uuid reference because other
-    // related models may be still using the temp uuid in their relationships
-    // data object. However, when we are listing out records we want them
-    // to be unique by the persisted id (which is updated after a Model.save)
+    const records = Array.from(this.getType(type).records.values()).filter(
+      (value) => value && value !== 'undefined'
+    )
     return uniqueBy(records, 'id')
   }
 
+  /**
+   * Get multiple records by id
+   *
+   * @method getRecordsById
+   * @param {String} type
+   * @param {Array} ids
+   * @return {Array} array or records
+   */
+  getRecordsById (type, ids = []) {
+    // NOTE: Is there a better way to do this?
+    return ids
+      .map((id) => this.getRecord(type, id))
+      .filter((record) => record)
+      .filter((record) => typeof record !== 'undefined')
+  }
+
```

Comment:
Isn't this repeating the same idea as `getMany`?

Reply:
This one doesn't look at cache at all - definitely apart of the getRecord weirdness that needs to be sorted out though yes, just not in these PRs